EventSubscriber/EventSubscriberInterface.php

```diff
@@ -2,8 +2,16 @@
 
 namespace Doctrine\Bundle\DoctrineBundle\EventSubscriber;
 
+use Doctrine\Bundle\DoctrineBundle\Attribute\AsDoctrineListener;
 use Doctrine\Common\EventSubscriber;
 
+use function trigger_deprecation;
+
+trigger_deprecation('doctrine/doctrine-bundle', '2.10', 'The "%s" class is deprecated, use "%s" attribute instead.', EventSubscriberInterface::class, AsDoctrineListener::class);
```

Comment:
no need for this deprecation triggered at the time of autoloading the class. The proper deprecation will be triggered by DebugClassLoader when loading a class that implements the interface based on the `@deprecated` phpdoc tag.

Reply:
ah ok thanks 🙏